torchx/lib/torchx/backend.ex

```diff
@@ -69,6 +69,29 @@ defmodule Torchx.Backend do
   defp from_torch_type(:float), do: {:f, 32}
   defp from_torch_type(:double), do: {:f, 64}
 
+  def device_option(opts), do: (opts[:backend_options] || [])[:device] || opts[:device] || :cpu
```

Comment:
This should be:

```suggestion
  def device_option(opts), do: opts[:device] || :cpu
```

And then you should do `opts[:backend_options]` on all places except the backend_transfer function. :)